pkgs/development/libraries/nghttp2/default.nix

```
@@ -1,82 +1,102 @@
+# should be included directly in Nixpkgs as files.
 
-assert enableHpack -> jansson != null;
+assert enableApp -> c-ares != null && libev != null && openssl != null && zlib != null;
 assert enableAsioLib -> boost != null;
-assert enableGetAssets -> libxml2 != null;
-assert enableJemalloc -> jemalloc != null;
+assert enableGetAssets -> enableApp == true && libxml2 != null;
+assert enableHpack -> enableApp == true && jansson != null;
+assert enableJemalloc -> enableApp == true && jemalloc != null;
 assert enablePython -> python != null && cython != null && ncurses != null && setuptools != null;
-
-let inherit (lib) optional optionals optionalString; in
+assert enableTests -> cunit != null && tzdata != null;
 
 stdenv.mkDerivation rec {
   pname = "nghttp2";
-  version = "1.43.0";
+  version = "1.47.0";
 
   src = fetchurl {
     url = "https://github.com/${pname}/${pname}/releases/download/v${version}/${pname}-${version}.tar.bz2";
-    sha256 = "0qhgyphzdv72dgdfxin2xbk9623za3jwbcvhhaxixiwp6djj8vsm";
+    sha256 = "11d6w8iqrhnxmjd9ss9fzf66f7a32d48h2ihyk1580lg8d3rkj07";
   };
 
   outputs = [ "bin" "out" "dev" "lib" ]
-    ++ optional enablePython "python";
+    ++ lib.optionals (enablePython) [ "python" ];
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ openssl ]
-    ++ optional enableLibEv libev
-    ++ [ zlib ]
-    ++ optional enableCAres c-ares
-    ++ optional enableHpack jansson
-    ++ optional enableAsioLib boost
-    ++ optional enableGetAssets libxml2
-    ++ optional enableJemalloc jemalloc
-    ++ optionals enablePython [ python ncurses setuptools ];
+  nativeBuildInputs = [ pkg-config ]
+    ++ lib.optionals (enableApp) [ installShellFiles ]
+    ++ lib.optionals (enablePython) [ cython ];
+
+  buildInputs = lib.optionals enableApp [ c-ares libev openssl zlib ]
+    ++ lib.optionals (enableAsioLib) [ boost ]
+    ++ lib.optionals (enableGetAssets) [ libxml2 ]
+    ++ lib.optionals (enableHpack) [ jansson ]
+    ++ lib.optionals (enableJemalloc) [ jemalloc ]
+    ++ lib.optionals (enablePython) [ python ncurses setuptools ];
 
   enableParallelBuilding = true;
 
   configureFlags = [
-    "--with-spdylay=no"
     "--disable-examples"
     (lib.enableFeature enableApp "app")
-  ] ++ optional enableAsioLib "--enable-asio-lib --with-boost-libdir=${boost}/lib"
-    ++ (if enablePython then [
-    "--with-cython=${cython}/bin/cython"
-  ] else [
-    "--disable-python-bindings"
-  ]);
+  ] ++ lib.optionals (enableAsioLib) [ "--enable-asio-lib" "--with-boost-libdir=${boost}/lib" ]
+    ++ lib.optionals (enablePython) [ "--with-cython=${cython}/bin/cython" ];
+
+  # Unit tests require CUnit and setting TZDIR environment variable
+  doCheck = enableTests;
+  checkInputs = lib.optionals (enableTests) [ cunit tzdata ];
+  preCheck = lib.optionalString (enableTests) ''
+    export TZDIR=${tzdata}/share/zoneinfo
+  '';
 
-  preInstall = optionalString enablePython ''
+  preInstall = lib.optionalString (enablePython) ''
```

```suggestion
  preInstall = lib.optionalString enablePython ''
```
Those are optional and where not there before.

---

It's my _personal preference_ to consistently surround the condition being tested with parentheses. I find it helps readability.